### PR TITLE
commonlib: added solution to 'Brute Forcing Credit Card Information' vulnerability

### PR DESCRIPTION
**Pull request by zuhairabid**
## Overview
Added solution for the vulnerabilty : 'Brute Forcing Credit Card Information'

## Related Issues
zaproxy/zaproxy#8056

### CODE REVIEW

File: addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml [hunk 1]
```diff
@@ -390,7 +390,25 @@ In order to fill in the missing information the hacker can guess the missing inf
...
+* CAPTCHA Protection: Implement CAPTCHA challenges during the login or payment process to differentiate between human and automated attempts. CAPTCHAs can help prevent automated brute force attacks.
+* Rate Limiting: Implement rate limiting on login or payment attempts. Limit the number of requests from a single IP address or user account within a specific time frame.
+* Multi-Factor Authentication (MFA): Require multi-factor authentication for sensitive transactions, including online payments. MFA adds an extra layer of security by verifying the user's identity through multiple factors, making brute force attacks more difficult.
+* Secure Data Storage: Ensure that credit card information is securely stored and encrypted. Limit access to sensitive data to authorized personnel only.
+* Continuous Monitoring: Implement continuous monitoring to detect and respond to suspicious activities, including multiple failed login attempts.
+* Compliance with PCI DSS: If your application processes credit card information, ensure compliance with Payment Card Industry Data Security Standard (PCI DSS) requirements, which provide security guidelines for handling credit card data.
```
Comment (kingthorin) [inline addOns/commonlib/src/main/resources/org/zaproxy/addon/commonlib/internal/vulns/vulnerabilities.xml:403]:
```suggestion
* CAPTCHA Protection: Implement CAPTCHA challenges during the payment process to differentiate between human and automated attempts. CAPTCHAs can help prevent automated brute force attacks.
* Rate Limiting: Implement rate limiting on payment attempts. Limit the number of requests from a single IP address or user account within a specific time frame.
* Multi-Factor Authentication (MFA): Require multi-factor authentication for sensitive transactions, including online payments. MFA adds an extra layer of security by verifying the user's identity through multiple factors, making brute force attacks more difficult.
* Secure Data Storage: Ensure that credit card information is securely stored and encrypted. Limit access to sensitive data to authorized personnel only.
* Continuous Monitoring: Implement continuous monitoring to detect and respond to suspicious activities, including multiple failed payment attempts.
* Compliance with PCI DSS: If your application processes credit card information, ensure compliance with Payment Card Industry Data Security Standard (PCI DSS) requirements, which provide security guidelines for handling credit card data.
```

**Comment by kingthorin** [comment]
To address the DCO requirement you'll need to sign-off the commit(s):

- https://github.com/zaproxy/zaproxy/blob/main/CONTRIBUTING.md#developer-certificate-of-origin
- https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff

**Comment by zuhairabid** [comment]
@kingthorin Thank you for your guideline can you have another look?

**Comment by kingthorin** [comment]
I'm not sure what you changed, but my recommendation is to remove the login controls and mentions. The vuln is specific to credit card info.

**Review by kingthorin — APPROVED** [review]

File: addOns/commonlib/CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - A generic UI component for keeping menu items sorted.
 
 ### Changed
+- Add solution to 'Brute Forcing Credit Card Information' vulnerability (Issue 8056).
+
```
Comment (thc202) [inline addOns/commonlib/CHANGELOG.md:13]:
```suggestion
```

File: addOns/commonlib/CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - A generic UI component for keeping menu items sorted.
 
 ### Changed
+- Add solution to 'Brute Forcing Credit Card Information' vulnerability (Issue 8056).
 - Add solution to 'Content Spoofing', 'Credential and Session Prediction', 'XML Injection' and 'XML External Entities' vulnerabilities (Issue 8056).
```
Comment (thc202) [inline addOns/commonlib/CHANGELOG.md:13]:
```suggestion
- Add solution to 'Brute Forcing Credit Card Information', 'Content Spoofing', 'Credential and Session Prediction', 'XML Injection' and 'XML External Entities' vulnerabilities (Issue 8056).
```

**Review by kingthorin — APPROVED** [review]

**Comment by ricekot** [comment]
Thanks!

**Comment by zuhairabid** [comment]
@ricekot Thank you! This was my first open source contribution. Feeling very happy

**Comment by thc202** [comment]
Thank you @zuhairabid! How would you like to be credited (e.g. name, handle)?
https://www.zaproxy.org/docs/desktop/credits/#zap-extended-team

**Comment by zuhairabid** [comment]
@thc202 If full name then  `Zuhair Abid` else `Zuhair` would be nice, thank you

**Comment by zuhairabid** [comment]
@thc202 for the handle @zuhairabid to my github profile link, thank you.

**Pull request merged**